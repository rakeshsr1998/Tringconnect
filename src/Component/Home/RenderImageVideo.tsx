import React, {useState} from 'react';
import {Image, Button, Text, View, Pressable, Dimensions} from 'react-native';
import Modal from 'react-native-modal';
import {styles} from './activity-style';
import Carousel from 'react-native-snap-carousel';

const width = Dimensions.get('screen').width * 0.9;

export const RenderImageOrVideo = ({item}: any) => {
  const [isModalVisible, setModalVisible] = useState(false);
  const [index, setIndex] = useState(0);
  const toggleModal = index => {
    setIndex(index);
    setModalVisible(!isModalVisible);
  };

  const render = ({item}: any) => {
    return (
      <Image
        resizeMode="cover"
        source={{uri: item}}
        style={styles.modalCover}
      />
    );
  };

  return (
    <>
      {item.images.length == 1 && (
        <Pressable onPress={() => toggleModal(0)} testID="image1">
          <Image source={{uri: item.images[0]}} style={styles.postImage} />
        </Pressable>
      )}
      {item.images.length > 1 && (
        <>
          <View style={styles.flexRow}>
            <Pressable
              style={styles.flexOne}
              onPress={() => toggleModal(0)}
              testID="image2">
              <Image source={{uri: item.images[0]}} style={styles.multiImage} />
            </Pressable>
            <Pressable
              style={styles.flexTwo}
              onPress={() => toggleModal(1)}
              testID="image3">
              <Image source={{uri: item.images[1]}} style={styles.multiImage} />
            </Pressable>
          </View>
          <View style={styles.flexRow}>
            <Pressable
              style={styles.flexThree}
              onPress={() => toggleModal(2)}
              testID="image4">
              <Image source={{uri: item.images[2]}} style={styles.multiImage} />
            </Pressable>
            <Pressable
              style={styles.flexFour}
              onPress={() => toggleModal(3)}
              testID="image5">
              <Image source={{uri: item.images[3]}} style={styles.multiImage} />
            </Pressable>
          </View>
        </>
      )}
      <Modal
        onBackdropPress={() => setModalVisible(false)}
        hasBackdrop={true}
        isVisible={isModalVisible}>
        <View>
          <Pressable
            testID='close'
            onPress={() => setModalVisible(false)}
            style={styles.closeIcon}>
            <Image
              source={require('../../assets/close.png')}
              style={styles.more}
            />
          </Pressable>
          <Carousel
            firstItem={index}
            pagingEnabled={true}
            data={item.images}
            renderItem={render}
            sliderWidth={width}
            itemWidth={width}
          />
        </View>
      </Modal>
    </>
  );
};
