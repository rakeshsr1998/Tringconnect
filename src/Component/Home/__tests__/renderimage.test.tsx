import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {RenderImageOrVideo} from '../RenderImageVideo';

const data = {
  id: 1,
  name: 'Kreethiraj K',
  role: 'Product Delivery Manager',
  post: 'Successfully delivered this great kitchen app to market. Kudos the entire team.',
  date: '2017-01-01',
  time: '12:34:56',
  likes: 240,
  comments: 12,
  images: [
    'https://images.unsplash.com/photo-1564758596151-c17d1590f483?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8cGhvbmUlMjBmb29kfGVufDB8fDB8fA%3D%3D&w=1000&q=80',
  ],
};

const data1 = {
  id: 1,
  name: 'Kreethiraj K',
  role: 'Product Delivery Manager',
  post: 'Successfully delivered this great kitchen app to market. Kudos the entire team.',
  date: '2017-01-01',
  time: '12:34:56',
  likes: 240,
  comments: 12,
  images: [
    'https://images.unsplash.com/photo-1564758596151-c17d1590f483?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8cGhvbmUlMjBmb29kfGVufDB8fDB8fA%3D%3D&w=1000&q=80',
    'https://images.unsplash.com/photo-1564758596151-c17d1590f483?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8cGhvbmUlMjBmb29kfGVufDB8fDB8fA%3D%3D&w=1000&q=80',
    'https://images.unsplash.com/photo-1564758596151-c17d1590f483?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8cGhvbmUlMjBmb29kfGVufDB8fDB8fA%3D%3D&w=1000&q=80',
    'https://images.unsplash.com/photo-1564758596151-c17d1590f483?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8cGhvbmUlMjBmb29kfGVufDB8fDB8fA%3D%3D&w=1000&q=80',
  ],
};

describe('<Render image in screen page>', () => {
  it('Should render one image', async () => {
    const image1 = render(<RenderImageOrVideo data={data} />).getByTestId(
      'image1',
    );
    fireEvent(image1, 'onPress');

    const image2 = render(<RenderImageOrVideo data={data1} />).getByTestId(
      'image2',
    );
    fireEvent(image2, 'onPress');

    const image3 = render(<RenderImageOrVideo data={data1} />).getByTestId(
      'image3',
    );
    fireEvent(image3, 'onPress');

    const image4 = render(<RenderImageOrVideo data={data1} />).getByTestId(
      'image4',
    );
    fireEvent(image4, 'onPress');

    const image5 = render(<RenderImageOrVideo data={data1} />).getByTestId(
      'image5',
    );
    fireEvent(image5, 'onPress');
  });
});
