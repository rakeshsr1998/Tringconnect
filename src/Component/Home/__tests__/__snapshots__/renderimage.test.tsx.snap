// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Render image in screen page> Should render 4 image 1`] = `[MockFunction]`;

exports[`<Render image in screen page> Should render 4 image 2`] = `[MockFunction]`;

exports[`<Render image in screen page> Should render 4 image 3`] = `[MockFunction]`;

exports[`<Render image in screen page> Should render 4 image 4`] = `[MockFunction]`;

exports[`<Render image in screen page> Should render 4 image 5`] = `[MockFunction]`;

exports[`<Render image in screen page> Should render header with comments like click 1`] = `[MockFunction]`;

exports[`<Render image in screen page> Should render one image 1`] = `[MockFunction]`;
